actors/builtin/multisig/multisig_actor.go

```diff
@@ -120,8 +117,14 @@ func (a *MultiSigActor) Propose(rt vmr.Runtime, params *ProposeParams) TxnID {
 		Value:      params.Value,
 	}
 
-	st.PendingTxns[txnID] = txn
 	st.PendingApprovals[txnID] = autil.ActorIDSetHAMT_Empty()
+
+	pendingTxnCID, err := setPendingTxn(context.TODO(), rt, txnID, txn)
+	if err != nil {
+		rt.AbortStateMsg(fmt.Sprintf("failed to set transaction in HAMT: %v", err))
+	}
```

Comment:
Push the error check and abort down into the helper method. And for getPendingTxn and deletePendingTxn too